streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -268,9 +274,13 @@ public void onPartitionsAssigned(final Collection<TopicPartition> assignment) {
             final long start = time.milliseconds();
             try {
                 if (streamThread.setState(State.PARTITIONS_ASSIGNED) == null) {
-                    return;
-                }
-                if (streamThread.assignmentErrorCode.get() == StreamsPartitionAssignor.Error.NONE.code()) {
+                    log.error("State transition from {} to PARTITIONS_ASSIGNED failed. " +
+                                  "Will skip the task initialization", streamThread.state());
+                } else if (streamThread.assignmentErrorCode.get() != StreamsPartitionAssignor.Error.NONE.code()) {
```

Comment:
Isn't this a rather brittle fix? How does this code guarantee that after `streamThread.assignmentErrorCode.get() != StreamsPartitionAssignor.Error.NONE.code()` is `false`, the result of `streamThread.assignmentErrorCode.get()` does not change again before the tasks are created?

Reply:
in `PartitionsAssigned` we should have already passed the assignment error placement, so I won't expect another flip for its value during this round rebalance.